qiskit_experiments/calibration_management/calibrations.py

```diff
@@ -89,15 +89,24 @@ def __init__(self, control_config: Dict[Tuple[int, ...], List[ControlChannel]] =
         self._hash_to_counter_map = {}
         self._parameter_counter = 0
 
-    def add_schedule(self, schedule: ScheduleBlock, qubits: Union[int, Tuple[int, ...]] = None):
+    def add_schedule(
+        self,
+        schedule: ScheduleBlock,
+        qubits: Union[int, Tuple[int, ...]] = None,
+        n_qubits: Optional[int] = None,
```

Comment:
I think I like `(qubits=(), n_qubits=2)` better than `(qubits=(-1, -1))` for tracking default schedules. It is more explicit and avoids using magic values.

The extra bookkeeping for n_qubits is kind of annoying though. In [this comment](https://github.com/Qiskit/qiskit-experiments/pull/386/files#r708987817), it was pointed out that schedule name is associated with a unique value for `n_qubits`. Could we just keeping a separate `schedule_name:n_qubits` mapping internally and leave the schedule key as `(name, qubits)`?

Also, I wonder how it would work out if we tracked `n_qubits` in the schedule metadata. It seems like an inherent property of the schedule that would be good to keep with the schedule instead of carrying around separately, but it might be an extra burden on users to remember to attach the metadata. Perhaps no more than remembering to specify `n_qubits` when calling `add_schedule` though.

Reply:
I agree. The extra bookkeeping is annoying. I introduced an extra mapping as you suggested, see here: 601e193 I went for this rather than the metadata. Think it makes the code's API a bit easier to understand than using the schedule's metadata.